From 680683deb765ca5ccf4624f4f04c52c2d0f61f34 Mon Sep 17 00:00:00 2001
From: Michael Ludwig <michaelludwig@google.com>
Date: Tue, 4 Aug 2026 14:39:10 -0400
Subject: [PATCH] [M150] [ganesh] Check result inside
 GrResourceProvider::writePixels

Original change's description:
> [ganesh] Check result inside GrResourceProvider::writePixels
>
> Bug: 540027341
> Fixed: 540027341
> Change-Id: Icde9d077dfb3ec3cdb7a6e775c6472443dba8781
> Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1315336
> Auto-Submit: Michael Ludwig <michaelludwig@google.com>
> Reviewed-by: Robert Phillips <robertphillips@google.com>
> Commit-Queue: Robert Phillips <robertphillips@google.com>

(cherry picked from commit 5e4f3217c17a8d88195d4c2924613420a256457b)

Bug: 546215390,540027341,540027341
Change-Id: Icde9d077dfb3ec3cdb7a6e775c6472443dba8781
Reviewed-on: https://skia-review.googlesource.com/c/skia/+/1325856
Auto-Submit: Chrome Cherry Picker <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
---
 src/gpu/ganesh/GrResourceProvider.cpp | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/third_party/skia/src/gpu/ganesh/GrResourceProvider.cpp b/third_party/skia/src/gpu/ganesh/GrResourceProvider.cpp
index f7598cb89d..b10b0a358e 100644
--- a/third_party/skia/src/gpu/ganesh/GrResourceProvider.cpp
+++ b/third_party/skia/src/gpu/ganesh/GrResourceProvider.cpp
@@ -922,11 +922,13 @@ sk_sp<GrTexture> GrResourceProvider::writePixels(sk_sp<GrTexture> texture,
     if (tempColorType == GrColorType::kUnknown) {
         return nullptr;
     }
-    SkAssertResult(fGpu->writePixels(texture.get(),
-                                     SkIRect::MakeSize(baseSize),
-                                     colorType,
-                                     tempColorType,
-                                     tmpTexels.get(),
-                                     mipLevelCount));
+    if (!fGpu->writePixels(texture.get(),
+                           SkIRect::MakeSize(baseSize),
+                           colorType,
+                           tempColorType,
+                           tmpTexels.get(),
+                           mipLevelCount)) {
+        return nullptr;
+    }
     return texture;
 }
-- 
2.56.0

